src/modules/Overlay/Module.js

```diff
@@ -47,20 +49,29 @@ module.exports = function Dynamic(options, UI, util) {
 
       function changePixel(r1, g1, b1, a1, x, y) {
 
+        const firstImagePixels = [r1, g1, b1, a1];
+
         // overlay
-        var p = options.secondImagePixels;
+        const p = options.secondImagePixels;
         if (x >= options.x
           && x - options.x < p.shape[0]
           && y >= options.y
-          && y - options.y < p.shape[1])
-          return [
+          && y - options.y < p.shape[1]){
+
+          const secondImagePixels = [
             p.get(x - options.x, y - options.y, 0),
             p.get(x - options.x, y - options.y, 1),
             p.get(x - options.x, y - options.y, 2),
             p.get(x - options.x, y - options.y, 3)
           ];
+          
+          if(secondImagePixels[3] === 0)
+            return firstImagePixels;
```

Comment:
So does this only allow binary transparency, no 50% alpha for example? Thank you!!!

Reply:
This will return the first image if the 2nd image is completely transparent, to reduce processing. For all non-0 values, it is going to overlay. (if I am correct)

Reply:
oh makes total sense, thanks!